app/models/concerns/orchestration/compute.rb

```diff
@@ -246,12 +263,17 @@ def find_address
       # images do require an IP, but it's more accurate to return something here
       # if we have it, and let the SSH orchestration fail (and notify) for an
       # unreachable IP
-      ip = vm.ip_addresses.first if ip.blank?
-      logger.info "acquisition of ip address timed out, using #{ip}"
+      ip = filter_ip_addresses(vm.ip_addresses, type).first if ip.blank?
+      logger.info "acquisition of #{type} address timed out, using #{ip}"
     end
     ip
   end
 
+  def filter_ip_addresses(addresses, type)
+    check_method = type == :ip6 ? :ipv6? : :ipv4?
+    addresses.map {|ip| IPAddr.new(ip) rescue nil}.compact.select { |ip| ip.send(check_method) }.map(&:to_s)
```

Comment:
`select { |ip| ip.send(check_method) }` can be written as `select(&check_method)`